library/std/src/sys/unix/fs.rs

```diff
@@ -313,8 +313,11 @@ pub struct FilePermissions {
     mode: mode_t,
 }
 
-#[derive(Copy, Clone)]
-pub struct FileTimes([libc::timespec; 2]);
+#[derive(Copy, Clone, Debug, Default)]
+pub struct FileTimes {
+    accessed: Option<SystemTime>,
+    modified: Option<SystemTime>,
```

Comment:
I specifically hoped to *avoid* making these use `Option`, as `SystemTime` doesn't have a niche, and this will make the struct substantially larger. I expect that some applications will have a need to generate and (temporarily) store a large number of these structures.

Reply:
If necessary, I would rather see the setters for `FileTimes` return `Result` than store an `Option` in `FileTimes` itself.

Reply:
@beetrees Thank you! If `SystemTime` moves to having a niche, then I'm entirely in favor of using `Option` here.